draft-ietf-httpbis-http2bis.xml

```diff
@@ -606,6 +600,54 @@ HTTP Frame {
           connection with a <xref target="ConnectionErrorHandler">connection error</xref> of type
           <xref target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref> if it does not decompress a field block.
         </t>
+        <t>
+          A decoding error in a field block MUST be treated as a <xref
+          target="ConnectionErrorHandler">connection error</xref> of type <xref
+          target="COMPRESSION_ERROR" format="none">COMPRESSION_ERROR</xref>.
+        </t>
+
+        <section anchor="dynamic-table">
+          <name>Compression State</name>
+          <t>
+            Field compression is stateful.  Each endpoint has an encoder context and a decoder
+            context that are used for encoding and decoding all field blocks.  <xref
+            target="COMPRESSION" section="4"/> defines the dynamic table, which is the primary state
+            for each context.
+          </t>
+          <t>
+            The dynamic table has a maximum size that is set by a decoder using the
```

Comment:
This is ludicrous nitpicking, but we're wordsmithing here so I want to bring it up.

I think we're using the words "encoder" and "decoder" here to mean two different things. In the first paragraph of this subsection you said:

> Field compression is stateful.  Each endpoint has an encoder context and a decoder context that are used for encoding and decoding all field blocks. This implies that "encoder" means "HPACK encoder" and "decoder" means "HPACK decoder".

Ok, great. Two endpoints, each with an encoder context and a decoder context.

Then, this paragraph (and several following) say things like:

> The dynamic table has a maximum size that is set by a decoder using the SETTINGS_HEADER_TABLE_SIZE setting

and 

> Any change to the maximum value set by the decoder using SETTINGS_HEADER_TABLE_SIZE takes effect when the encoder acknowledges settings.

To read _this_ sentence correctly I think I have to understand "encoder" and "decoder" to refer to _endpoints_. That is, "encoder" refers to "the endpoint encoding headers" and "decoder" refers to "the endpoint decoding headers". After all, HPACK decoders don't know anything about SETTINGS_HEADER_TABLE_SIZE, and HPACK encoders don't acknowledge settings.

This introduces terminology that the section hasn't used before: the idea of an implicit "encoder" role, whose definition is (I think) "the endpoint transforming headers from a semantic field block to an encoded HPACK block". Awkwardly, of course, an encoder in this definition also has an encoder (and a decoder!) in the definition of the first paragraph of the section.

We then flip back and forth between these two uses of the word "encoder" in the same paragraph. For example, paragraph 3 uses both senses of the words (notes in braces and bolded added by me), occasionally even in the same sentence:

> Any change to the maximum value set by the decoder **{decoder as endpoint}** using SETTINGS_HEADER_TABLE_SIZE takes effect when the encoder acknowledges settings. The encoder **{encoder as HPACK encoder}** can set the dynamic table to any size up to the maximum value set by the decoder **{decoder as endpoint}**.  The encoder **{encoder as HPACK encoder, but maybe also as endpoint?}** declares the size of the dynamic table with a Dynamic Table Size Update instruction.

I don't think this flexibility with language is helping to clarify a very murky area. I think we need to purge these paragraphs of any use of the words "encoder" or "decoder" that cannot sensibly have the word "HPACK" written directly in front of them. Perhaps we should just always write "HPACK encoder" and "HPACK decoder" to avoid ambiguity?

Reply:
I think the clearest way might be to mention the endpoints explicitly, something like "The dynamic table has a maximum size that is set by the decoder.  An endpoint communicates the size chosen by its decoder context using the SETTINGS_HEADER_TABLE_SIZE setting...."

That is, only endpoints send settings while HPACK encoders only produce header blocks.